field_utils/mapl_udunits2cfunc.c

```diff
@@ -0,0 +1,16 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "udunits2.h"
+
+/* Helper function to augment udunits2 error handling
+ * Sets the udunits2 error handler to ut_ignore
+ * which disables error messages from udunits2
+ * udunits2 requires a ut_error_message_handler be passed
+ * into ut_set_error_message_handler to change the error handler,
+ * and ut_error_message_handler is a function with a variadic list
+ * of arguments, which is not possible in Fortran.
+*/
+ut_error_message_handler ut_set_ignore_error_message_handler()
+{
+    return ut_set_error_message_handler(ut_ignore);
+}
```

Comment:
@darianboggs Not that there is a problem with including C code here, but ... why couldn't you just add an interoperable interface in `mapl_udunits2interfaces.F90` and then called it in the top level initialize?

Reply:
Seems like what we would want to do for the external package.  It then allows a Fortran user to specify other error handling -not that we expect to do so.

Reply:
Oh  - I see now that `ut_ignore` is a func not an enum.   That might make the Fortran interface a bit difficult - outside of my ken ...   But probably worth exploring.

Reply:
I agree that it would be better to use `mapl_udunits2interfaces.F90`.

Turning off the output requires 2 functions. `ut_set_error_message_handler` sets the error_message_handler, and `ut_ignore` is a `ut_error_message_handler` that is passed in. Unfortunately, `ut_error_message_handler` instances have variadic arguments, like `printf`. I don't think the Fortran interface to C allows functions with variadic arguments. The default `ut_error_message_handler` sends all messages to STDERR.

So while we want to use the `ut_ignore` handler, which ignores all messages and does nothing, I can't write an interface for `ut_ignore`. I thought about using a `c_funptr` in the interface for `ut_set_error_message_handler`, but I would still need to pass in `ut_ignore` as the `c_funptr`.

Reply:
ok - thanks for the reminder.

Reply:
abstract interface
    
    subroutine ut_set_error_message_handler(handler) bind(c, name="ut_set_error_message_handler")
      import :: c_funptr
      type(c_funptr), intent(in) :: handler
    end subroutine ut_set_error_message_handler

    subroutine ut_ignore(fmt, args) bind(c, name="ut_ignore")
      import :: ??? ! not sure what to use
      character(kind=c_char), intent(in) :: fmt(*)
      type(va_list), intent(in) :: args ! Can't do this
    end subroutine ut_ignore
    
    end interface

Reply:
Resolving for now to allow merge.